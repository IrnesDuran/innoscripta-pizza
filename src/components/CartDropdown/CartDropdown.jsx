import React from 'react';
import {connect} from 'react-redux';
import './CartDropdown.styles.scss';
import CustomButton  from '../../components/CustomButton/CustomButton'
import {withRouter} from 'react-router-dom';

import CartItem from '../CartItem/CartItem';

import  { toggleCartHidden } from '../../redux/cart/cart.actions';



export const CartDropdown =(props) => {
    
    const {cartItems, history, dispatch} = props;

    return(
    <div className='cart-dropdown'>
        <div className='cart-items'>
            {

            cartItems.length ? 

            cartItems.map(cartItem => 
                 <CartItem 
                 key= {cartItem.id}
                 item={cartItem}
                 />
                ) : <span className= 'empty-message'>Your cart is empty</span>
            }

        </div>
        <CustomButton onClick= {() => {
            history.push('/checkout');
            dispatch(toggleCartHidden());

            }}>Go to checkout</CustomButton>
    </div>


)};
const mapStateToProps = state => ({
    cartItems: state.cart.cartItems
  });

export default withRouter(connect(mapStateToProps)(CartDropdown));