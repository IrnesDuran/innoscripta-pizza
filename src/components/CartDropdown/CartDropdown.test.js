import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });


import {CartDropdown} from './CartDropdown';




describe('Testing CartDropdown',()=>{

    let mockProps;
    let wrapper;

    beforeEach(() => {
        mockProps = {cartItems:[]}
        wrapper = shallow(<CartDropdown {...mockProps}/>);
    });
    

 

    it('should test CartDropdown snapshot',()=>{
        expect(wrapper).toMatchSnapshot();
    });

    it('should test if location pathname prior to being included in Checkout Page', () =>{
        expect(location.pathname).toEqual('/');

    });

});

