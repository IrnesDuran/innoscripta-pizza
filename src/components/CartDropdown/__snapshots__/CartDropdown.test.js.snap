// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CartDropdown should test CartDropdown snapshot 1`] = `ShallowWrapper {}`;
